react-validated-input-component/my-app/src/ValidatedInput.jsx

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCircleXmark } from '@fortawesome/free-regular-svg-icons';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
+
+export default function ValidateInput() {
+  const [password, setPassword] = useState('');
+  function handleChange(e) {
+    setPassword(e.target.value);
+  }
+
+  return (
+    <div>
+      <label>Password:</label>
+      <input type="password" value={password} onChange={handleChange} />
+      <span className="icon">
+        {password.length < 8 ? (
```

Comment:
For this challenge, having this logic in JSX works! 

However, what if we want to expand validation as per the Extra challenge? The code can get messy! Creating a `validate` function that returns an error message would be cleaner in that case! 

Sometimes JSX like you did is the best solution! Sometimes, separating it out is. That will come with experience :)  

We will go over that example during class on Monday!